packages/yarnpkg-core/sources/Project.ts

```diff
@@ -1548,6 +1548,26 @@ export class Project {
 
       await this.resolveEverything(opts);
 
+      for (const [, extensionsPerRange] of this.configuration.packageExtensions) {
+        for (const [, extensions] of extensionsPerRange) {
+          for (const extension of extensions) {
+            if (extension.userProvided) {
+              const prettyPackageExtension = formatUtils.pretty(this.configuration, extension, formatUtils.Type.PACKAGE_EXTENSION);
+
+              switch (extension.status) {
+                case `inactive`: {
+                  opts.report.reportWarning(MessageName.UNUSED_PACKAGE_EXTENSION, `Unused package extension: ${prettyPackageExtension}`);
+                } break;
+
+                case `redundant`: {
+                  opts.report.reportWarning(MessageName.UNNEEDED_PACKAGE_EXTENSION, `Unneeded package extension: ${prettyPackageExtension}`);
```

Comment:
Should it go into its own step, just to separate it from the "true" resolution errors (similar to how package validation is its own step)?

Reply:
🤔 I'm a bit worried about having too many steps. I put it inside the Resolution step because the immutable lockfile validation was also there (https://github.com/yarnpkg/berry/blob/a76373ac01680b85e6499dc2085abac00d4db575/packages/yarnpkg-core/sources/Project.ts#L1585).